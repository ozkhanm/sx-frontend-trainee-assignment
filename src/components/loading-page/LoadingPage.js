import React from "react";

import PageHeader from "../page-header/PageHeader";

const LoadingPage = () => {
    return (
        <React.Fragment>
            <PageHeader page={`MAIN_PAGE`}/>
            <h1 className="load">Загрузка данных...</h1>
        </React.Fragment>
    );
};

export default LoadingPage;