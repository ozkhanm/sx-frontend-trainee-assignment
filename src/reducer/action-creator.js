const ActionType = {
    GET_ARTICLES: `GET_ARTICLES`,
    CHANGE_LOADING_STATUS: `CHANGE_LOADING_STATUS`
};

const ActionCreator = {
    getArticles: (articles) => ({
        type: ActionType.GET_ARTICLES,
        payload: articles
    }),
    changeLoadingStatus: (status) => ({
        type: ActionType.CHANGE_LOADING_STATUS,
        payload: status
    })
};

export {ActionCreator, ActionType};