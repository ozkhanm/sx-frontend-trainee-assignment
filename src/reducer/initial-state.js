const initialState = {
    articles: [],
    isDataLoaded: false
};

export default initialState;