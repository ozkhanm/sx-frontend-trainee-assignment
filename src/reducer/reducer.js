import initialState from "./initial-state";
import {ActionType, ActionCreator} from "./action-creator";

const Operation = {
    getArticles: () => (dispatch, getState, api) => {
        return api.get(`/newstories.json?print=pretty`)
            .then((response) => {
                const ids = response.data.slice(0, 100);
                
                return Promise.all(ids.map((it) => api.get(`item/${it}.json?print=pretty`)));
            })
            .then((response) => {
                const articles = response.slice().map((it) => it.data);

                dispatch(ActionCreator.getArticles(articles));
            });
    }
};

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case ActionType.CHANGE_LOADING_STATUS:
            return Object.assign({}, state, {
                isDataLoaded: action.payload
            });

        case ActionType.GET_ARTICLES:
            return Object.assign({}, state, {
                articles: action.payload
            });
    }

    return state;
};

export {reducer, Operation};