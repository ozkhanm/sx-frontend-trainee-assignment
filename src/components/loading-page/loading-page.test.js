import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {createStore} from "redux";

import LoadingPage from "./LoadingPage";

import {reducer} from "../../reducer/reducer";

it(`Loading screen renders correctly`, () => {
    const tree = renderer
        .create(
            <Provider store={createStore(reducer)}>
                <LoadingPage/>
            </Provider>)
        .toJSON();
  
    expect(tree).toMatchSnapshot();
});