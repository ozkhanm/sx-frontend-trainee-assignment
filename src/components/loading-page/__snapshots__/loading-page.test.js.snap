// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading screen renders correctly 1`] = `
Array [
  <div>
    <div
      className="header-controls"
    >
      <a
        className="header-controls-button header-controls-refresh-button "
        onClick={[Function]}
      />
    </div>
    <p
      className="header"
    >
      Hacker News
    </p>
  </div>,
  <h1
    className="load"
  >
    Загрузка данных...
  </h1>,
]
`;
